frame/alliance/src/lib.rs

```diff
@@ -1167,4 +1175,25 @@ impl<T: Config<I>, I: 'static> Pallet<T, I> {
 		)?;
 		Ok(info.into())
 	}
+
+	/// Check invariants are consistent within the pallet.
+	/// - `Members(Fellows)` should keep to `MaxFellows` limit.
+	/// - `Members(Allies)` should keep to `MaxAllies` limit.
+	#[cfg(any(test, feature = "try-runtime", feature = "fuzz"))]
+	pub fn do_try_state() -> Result<(), &'static str> {
+		let fellows = Members::<T, I>::get(MemberRole::Fellow);
+		let allies = Members::<T, I>::get(MemberRole::Ally);
+
+		assert!(
+			fellows.len() as u32 <= T::MaxFellows::get(),
```

Comment:
it would be more correct to assert with `T::MaxMembersCount`.
and optionally check that `T::MaxFellows` and `T::MaxAllies` not greater than `T::MaxMembersCount`.

Reply:
The storage uses `T::MaxMembersCount` as a bound.
And `T::MaxFellows` and `T::MaxAllies` only used for the benchmarks.
Check the code bases, where these limits used, it should be clear.

Reply:
if `fellows.len() as u32 <= T::MaxFellows::get()` not true, there is a probability for the wrong wieghts.
if `fellows.len() as u32 <= T::MaxMembersCount::get()` not true, there is a wrong state.

Reply:
That's true, thank you for spotting that out!